crates/flycheck/src/lib.rs

```diff
@@ -538,7 +540,7 @@ impl CargoActor {
 }
 
 enum CargoMessage {
-    CompilerArtifact(cargo_metadata::Artifact),
+    CompilerArtifact(Box<cargo_metadata::Artifact>),
```

Comment:
This came up a couple of times before and I think bjorn3 pointed put that most of the time we're going to have this variant, so making this change will cause a lot of allocations.

I don't know if the cost of moving the large struct around is lower or higher than that of the allocations, though.

Reply:
I am fine with any choice. I was just going with the clippy recommendation here.

I'd propose that we go with it and if someone takes the time to compare the two options we potentially revert it and disable the clippy lint for it.

Reply:
https://github.com/rust-lang/rust-analyzer/pull/16470